src/librustc_platform_intrinsics/lib.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![cfg_attr(stage0, feature(custom_attribute))]
+#![crate_name = "rustc_platform_intrinsics"]
+#![unstable(feature = "rustc_private")]
+#![staged_api]
+#![crate_type = "dylib"]
+#![crate_type = "rlib"]
+#![feature(staged_api, rustc_private)]
+
+extern crate rustc_llvm as llvm;
+extern crate rustc;
+
+use rustc::middle::ty;
+
+pub struct Intrinsic {
+    pub inputs: Vec<Type>,
+    pub output: Type,
+
+    pub definition: IntrinsicDef,
+}
+
+#[derive(Clone)]
+pub enum Type {
+    Integer(u8),
+    Float(u8),
+    Pointer(Box<Type>),
+    Vector(Box<Type>, u8),
+}
```

Comment:
If this enum is just translated straight to a `Ty` below, perhaps this could just be a `Ty` in-memory and there could be an initialization pass for loading intrinsics into the tcx?